packages/adapter-squid/src/SquidAdapter.ts

```diff
@@ -0,0 +1,226 @@
+         success: false,
+         data: {},
+      };
+
+      const squid = this.squidProvider.getSquid();
+      const signer = this.signerService.getSignerForChain(
+         step.config.fromChain
+      );
+
+      let transaction;
+
+      try {
+         const route = await this.getRoute(step.config, squid);
+         transaction = await squid.executeRoute({ signer, route });
+      } catch (error) {
+         result.success = false;
+         result.data.errorMessage = `${ERR_ADAPTER_EXECUTE_FAILURE}:${
+            (error as Error).message
+         }`;
+         return result;
+      }
+
+      const txReceipt = await (
+         transaction as ethers.providers.TransactionResponse
+      ).wait();
+
+      if (txReceipt && txReceipt.transactionHash) {
+         result.success = true;
+         result.data.txHash = txReceipt.transactionHash;
+      } else {
+         result.success = false;
+         result.data.errorMessage = ERR_TX_RECEIPT_MISSING;
+      }
+
+      return result;
+   }
+
+   private async getRoute(
+      stepConfig: TypeOf<typeof BridgeSwapConfigCodec>,
+      squid: Squid
+   ): Promise<RouteData> {
+      const primaryAddress =
+         this.configProvider.getGlobalVariable("primaryAddress");
+      if (!primaryAddress) {
+         throw new Error(ERR_GLOBAL_VARIABLE_MISSING("primaryAddress"));
+      }
+
+      const getRouteParams = this.constructRouteParams(
+         stepConfig,
+         primaryAddress
+      );
+
+      return this.fetchRouteData(squid, getRouteParams);
+   }
+
+   private constructRouteParams(
+      stepConfig: TypeOf<typeof BridgeSwapConfigCodec>,
+      primaryAddress: string
+   ): GetRoute {
+      return {
+         fromChain: stepConfig.fromChain,
+         fromToken: stepConfig.fromToken,
+         fromAmount: stepConfig.fromAmount,
+         toChain: stepConfig.toChain,
+         toToken: stepConfig.toToken,
+         fromAddress: primaryAddress,
+         toAddress: stepConfig.toAddress,
+         slippage: stepConfig.maxSlippage,
+      };
+   }
+
+   private async fetchRouteData(
+      squid: Squid,
+      getRouteParams: GetRoute
+   ): Promise<RouteData> {
+      try {
+         const routeData = (await squid.getRoute(getRouteParams)).route;
+         if (!routeData) {
+            throw new Error(ERR_GET_ROUTE_UNDEFINED);
+         }
+         return routeData;
+      } catch (error) {
+         throw this.handleRouteError(error);
+      }
+   }
+
+   private handleRouteError(error: any): Error {
+      if (axios.isAxiosError(error)) {
+         const errorMessage = this.extractErrorMessages(error?.response?.data);
+         return new Error(`${ERR_GET_ROUTE_FAILURE}:${errorMessage}`);
+      }
+      return new Error(`${ERR_GET_ROUTE_FAILURE}:${error.message}`);
+   }
+
+   private extractErrorMessages(data: {
+      errors?: { message: string }[];
+   }): string {
+      if (!data?.errors || !Array.isArray(data.errors)) {
+         return UNKNOWN_SQUID_ERROR;
+      }
```

Comment:
I think this should do the same but slightly shorter: 

```suggestion
    return {
      ...stepConfig,
      fromAddress: primaryAddress,
    }
```

Reply:
looks like VS Code messed up the line numbers on this suggestion somehow, but you get the idea :)